components/global/mixer-adapter/pkg/publisher/publisher.go

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package publisher
+
+import (
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"net/http"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/cellery-io/mesh-observability/components/global/mixer-adapter/pkg/store"
+)
+
+type (
+	Publisher struct {
+		Ticker      *time.Ticker
+		Logger      *zap.SugaredLogger
+		SpServerUrl string
+		HttpClient  *http.Client
+		Persister   store.Persister
+	}
+)
+
+func (publisher *Publisher) Run(stopCh <-chan struct{}) {
+	publisher.Logger.Info("Publisher started")
+	for {
+		select {
+		case <-stopCh:
+			return
+		case <-publisher.Ticker.C:
+			publisher.execute()
+		}
+	}
+}
+
+func (publisher *Publisher) execute() {
+	for {
+		str, transaction, err := publisher.Persister.Fetch()
+		if err != nil {
+			publisher.Logger.Error(err.Error())
```

Comment:
Better to add some context here without logging only the error. Please do so in other error logs as well.